postgres.go

```diff
@@ -25,6 +27,93 @@ type DBConf struct {
 	clientKey  string
 }
 
+// Basebackup function:
+// - gets an identical copy of the pg database (pg_data)
+// - verifies the backup
+// - tars the copy
+// - compresses the encrypted file
+// - gets the key and encrypts the tar file
+// - puts the encrypted and compressed file in S3
+func (db DBConf) basebackup(sb s3Backend, keyPath string) error {
+	today := time.Now().Format("20060102150405")
+	destDir := "db-backup"
+	dbURI := buildConnInfo(db)
+	cmd := exec.Command("pg_basebackup", dbURI, "-F", "p", "-D", destDir)
+
+	var errMsg bytes.Buffer
+	cmd.Stderr = &errMsg
+
+	err := cmd.Run()
+	if err != nil {
+		log.Errorf(errMsg.String())
+		return err
+	}
+
+	cmd = exec.Command("pg_verifybackup", destDir)
+
+	cmd.Stderr = &errMsg
+
+	err = cmd.Run()
+	if err != nil {
+		log.Errorf(errMsg.String())
+		return err
+	}
+
+	cmd = exec.Command("tar", "-cvf", destDir+".tar", destDir)
+
+	cmd.Stderr = &errMsg
+
+	err = cmd.Run()
+	if err != nil {
+		log.Errorf(errMsg.String())
+		return err
+	}
+
+	fileName := today + "-" + db.database + ".enc"
+	wg := sync.WaitGroup{}
+	wr, err := sb.NewFileWriter(fileName, &wg)
+	if err != nil {
+		log.Errorf("Could not open backup file for writing: %v", err)
+		return err
+	}
+
+	key := getKey(keyPath)
+	e, err := newEncryptor(key, wr)
+	if err != nil {
+		log.Errorf("Could not initialize encryptor: (%v)", err)
+		return err
+	}
+
+	c, err := newCompressor(e)
+	if err != nil {
+		log.Errorf("Could not initialize compressor: (%v)", err)
+		return err
+	}
+
+	sourceFileName := destDir + ".tar"
+	data, err := ioutil.ReadFile(sourceFileName)
```

Comment:
```suggestion
	data, err := os.ReadFile(sourceFileName)
```
`ioutil` is deprecated now when we use GO > 1.16

Reply:
There are several problems with that on other tests etc and I think this is a job that can be done in the refactoring. Creating issue about that